proto/substrait/algebra.proto

```diff
@@ -178,6 +178,18 @@ message JoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+// A time-series variant of the multi JOIN relational operator left-join-right(s), which joins on an ordered key using inexact matching
+message AsOfJoinRel {
+  JoinRel join = 1;
```

Comment:
Does an AsOfJoinRel use all the properties of `join`?  In particular, does it use the `expression` or `type`?  If not, maybe composition is not the correct approach here and it would be better to just copy the relevant fields.

Reply:
I agree with Weston here, let's repeat the fields instead of sticking a join inside of another join.

As of joins are significantly different from a generic join that they should not be subject to any changes that happen in `JoinRel`.

Reply:
Yes and yes. See my earlier response.